src/components/ui4/DotsBackground.tsx

```diff
@@ -20,8 +20,9 @@ export function DotsBackground(props: Props): JSX.Element {
     accentDots[0] = { ...accentDots[0], color: accentColor }
   }
 
-  const [{ width, height }, setLayout] = React.useState({ width: 0, height: 0 })
+  const [{ width, height }, setLayout] = React.useState(cachedLayout)
   const handleLayout = useHandler((event: LayoutChangeEvent) => {
+    cachedLayout = event.nativeEvent.layout
```

Comment:
Could this safely be implemented in `useLayout`? I think so.

Reply:
Not really. I would totally use the `useLayout` hook, aside from this one problem.

Basically, every time we would transition from one scene to another, the background would glitch black as we waited for the layout to calculate. However, the calculation would always produce the same value, since we are full-screen on the phone every time. To solve the flash-of-black, I could either use `Dimensions.get(`window`)` as the default value for the layout, or I could use the last-seen value. I chose the last-seen value, which I cache in a global variable.

Since these assumptions are so specific to this one component, I don't see any way to incorporate them into the hook in a generic way.